iphone/TitaniumKit/TitaniumKit/Sources/API/TiExceptionHandler.m

```diff
@@ -156,8 +156,19 @@ - (id)initWithDictionary:(NSDictionary *)dictionary
     if (_backtrace == nil) {
       _backtrace = [[[dictionary objectForKey:@"stack"] description] copy];
     }
-    _nativeStack = [[dictionary objectForKey:@"nativeStack"] copy];
     _dictionaryValue = [dictionary copy];
+
+    id nativeStackObject = [dictionary objectForKey:@"nativeStack"];
+    if (nativeStackObject) {
+      if ([nativeStackObject isKindOfClass:[NSArray class]]) {
+        nativeStackObject = [nativeStackObject copy];
+      } else if ([nativeStackObject isKindOfClass:[NSString class]]) {
+        nativeStackObject = [[nativeStackObject componentsSeparatedByString:@"\n"] retain];
+      } else {
+        nativeStackObject = nil;
+      }
+    }
+    _nativeStack = nativeStackObject;
```

Comment:
Probably you can replace it with following -
```suggestion
    if ([nativeStackObject isKindOfClass:[NSString class]]) {
      nativeStackObject = [nativeStackObject componentsSeparatedByString:@"\n"];
    }

    if (![nativeStackObject isKindOfClass:[NSArray class]]) {
      nativeStackObject = nil;
    }
    _nativeStack = [nativeStackObject copy];
```

Reply:
What you're recommending does a needless copy if `nativeStackObject` is a string... and a stack trace string is normally quite large. My code retains the array created by the `componentsSeparatedByString` method. I still prefer the way I wrote it.

That said, I probably don't need the `if (nativeStackObject) {}` check since we can do `nil` message calls. So we can probably shorten it to this if we want.
```objc
id nativeStackObject = [dictionary objectForKey:@"nativeStack"];
if ([nativeStackObject isKindOfClass:[NSArray class]]) {
	nativeStackObject = [nativeStackObject copy];
} else if ([nativeStackObject isKindOfClass:[NSString class]]) {
	nativeStackObject = [[nativeStackObject componentsSeparatedByString:@"\n"] retain];
} else {
	nativeStackObject = nil;
}
_nativeStack = nativeStackObject;
```

Reply:
In your code you are copying nativeStackObject  in same variable but not releasing. You are retaining nativeStackObject, but not releasing.

_nativeStack is not retaining or copying  any object but it is getting released in dealloc. 
The code which I have suggested, it is also copying (owns memory) the nativeStackObject in _nativeStack.
Other way to do it is  -
`id nativeStackObject = [dictionary objectForKey:@"nativeStack"];
if ([nativeStackObject isKindOfClass:[NSArray class]]) {
	_nativeStack = [nativeStackObject copy];
} else if ([nativeStackObject isKindOfClass:[NSString class]]) {
	_nativeStack = [[nativeStackObject componentsSeparatedByString:@"\n"] retain];
} else {
	_nativeStack = nil;
}
//_nativeStack = nativeStackObject;`

Reply:
Updated this PR based on your suggestion.